### DISCUSSION TITLE
I need to share a folder with the students [SOLVED]

### DISCUSSION THREAD

**Discussion by Junglaverde** [ltsp/ltsp#435] [General]
Hello.
I need to share a server folder with clients. I want to put in some files and give clients access on boot.
The server has the address 192.168.67.1, the folder I want to share is in: /home/teacher/common

You can tell me the steps to follow and the commands. I have tried various configurations but can't do it.

Thanks a lot.

Set up
Ubuntu mate 20.04
two network cards
rootless

**Comment by alkisg** [comment]
LTSP doesn't manage folder sharing, so you may google for e.g. "share a network directory over NFS" for instructions.

In short, you may create an /etc/exports.d/local.exports file from where you'll export /home/teacher/common,
and you may mount it to the client's e.g. /mnt directory with an FSTAB line in ltsp.conf:

```shell
[clients]
FSTAB_MNT="server:/home/teacher/common /mnt nfs defaults,nolock 0 0"
```

**Reply by Junglaverde** [reply]
Hello again.
I have already managed to share the folder from the server to the clients. Then I write how I got it, in case someone needs to know how to do it.

**Reply by Junglaverde** [reply]
First we go to our directory and create the folder that we want to share. Then we give it permissions:
sudo chmod -R  755 /home/teacher/common

Then we edit the export file:
sudo nano /etc/exports.d/local.exports
And we add:
/home/teacher/common 192.168.67.0/24 (rw,sync,no_subtree_check)
Then we execute:
sudo exportfs -arv
sudo systemctl enable nfs-kernel-server
sudo systemctl start nfs-kernel-server

Finally, we add the path in ltsp.conf so that the client can mount it at the beginning.
sudo nano /etc/ltsp/ltsp.conf
And we add:
[clients]
  FSTAB_MNT="192.168.67.1:/home/teacher/common  /mnt  nfs  defaults,nolock 0 0"

Remember to do:
sudo ltsp ipxe
sudo ltsp initrd

We reboot and ready

(192.168.67.1 is the IP address of the client network on my server. You have to change it if necessary)

**Reply by Junglaverde** [reply]
Thank you very much Alkisg, without your guidance I would not have been able to do it.
